mitodl/open-discussions
# Fixed error page on comment error

#### What are the relevant tickets?
Fixes #477 

#### What's this PR do?
It hides the reply button for deleted posts, fixes the API to return a 410 status if you try to reply to one (instead of a 500), and updates the UI to show a toast message if you do so. Also has a catchall for any non-200 error so the user gets an error message and not an error page.

#### How should this be manually tested?
- Delete a comment, verify the reply button goes away
- Open the same post in two tabs:
  - delete a comment in one tab
  - without refreshing the page in the other tab, reply to that same comment and verify you get a 🍞 message

#### Screenshots (if appropriate)
![selection_123](https://user-images.githubusercontent.com/28598/35983262-ad9fe922-0cbf-11e8-858a-404bbfe2b99b.png)
